sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java

```diff
@@ -128,6 +140,16 @@ public void afterPropertiesSet() {
         }
     }
 
+    /**
+     * Validate common scenario properties configuration.
+     */
+    private void validateCommonProperties() {
+        if (!CollectionUtils.isEmpty(authorizationClients) && !StringUtils.hasText(tenantId)) {
```

Comment:
If we want to throw the exception if client credential flow is in the authorization clients, should we filter the authorization clients like:
```java
authorizationClients.filter(c -> ClientCredential...) ?
```

Reply:
In case we enable more OAuth 2.0 flow later, this logic now will break.